File: app/src/main/java/org/kexie/android/ftper/widget/DownloadWorker.java
package org.kexie.android.ftper.widget;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class DownloadWorker extends Worker {

    public DownloadWorker(@NonNull Context context,
                          @NonNull WorkerParameters workerParams) {
        super(context, workerParams);

    }

    @NonNull
    @Override
    public Result doWork() {
        return Result.failure();
    }
}

File: app/src/main/java/org/kexie/android/ftper/model/DownloadWorker.kt
package org.kexie.android.ftper.model

import android.content.Context
import androidx.work.ListenableWorker
import androidx.work.WorkerParameters

class DownloadWorker(context: Context, workerParams: WorkerParameters)
    : FTPWorker(context, workerParams) {

    override fun doWork(): ListenableWorker.Result {
        return try {
            connect()
            mConfig.file
            
            ListenableWorker.Result.success()
        } catch (e: Throwable) {
            e.printStackTrace()
            ListenableWorker.Result.failure()
        }
    }
}
